backend/src/layer/data/swka_parser/swka_resolver.rs

```diff
@@ -0,0 +1,88 @@
+use crate::interface::mensa_parser::ParseError;
+use futures::future::join_all;
+use reqwest::Client;
+use std::time::Duration;
+use tracing::log::debug;
+
+pub struct SwKaResolver;
+
+impl SwKaResolver {
+    pub const fn new() -> Self {
+        Self
+    }
+
+    fn get_client() -> Result<Client, ParseError> {
+        let client = Client::builder() // dont change or tests will fail!
+            .danger_accept_invalid_certs(true) //probably not needed
+            .timeout(Duration::from_millis(1000))
+            .connect_timeout(Duration::from_millis(1000))
+            .user_agent("User-Agent=Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.114 Safari/537.36")
+            .build();
+        match client {
+            Ok(cl) => Ok(cl),
+            Err(_e) => Err(ParseError::ClientBuilderFailed),
+        }
+    }
+
+    /// This function provides html code, which will be requested with the given url.<br>
+    /// If the request or the decoding fails an error will be thrown. <br>
+    /// `urls: Vec<String>` <br> urls to the requested html strings.<br>
+    /// **Return** <br> All html strings obtained by the urls as `Vec<String>`.
+    pub async fn get_html_strings(&self, urls: Vec<String>) -> Result<Vec<String>, ParseError> {
+        join_all(urls.iter().map(|url| self.get_html(url)))
```

Comment:
Should we remove the paralellism? we dont want to overload the website (which schould not happen from these few requests) and it is not time critical...